### DISCUSSION TITLE
Problem with real-time retweeting

### DISCUSSION THREAD

**Discussion by yekmil** [tweepy/tweepy#1534] [Questions] [answered]
I've been running my code locally for two years now. I run this code and it will retweet in under 15 minutes... This is a fast retweet bot and I run it manually with my IDE. 

I want to deploy it 24/7 on a server... with my calculation it needs 36 seconds pause between each tweet. The problem is when there is a 36 second sleep , the bot is no longer real-time. it will go the past. it ignores whatever is tweeting right now. I thought to include the cole insitde an infinite while loop but then that would be too slow because it should go through all the tweets all over again. 

Is there a way to resolve this issue? What do you propose?

```
import tweepy
from time import sleep
from random import randint
from credentials import *


# Access and authorize our Twitter credentials from credentials.py
auth = tweepy.OAuthHandler(consumer_key, consumer_secret)
auth.set_access_token(access_token, access_token_secret)
api = tweepy.API(auth)


since = '2021-01-28'
tag = 'keyword'
number = 1
# Looping through tweets
for tweet in tweepy.Cursor(api.search, q=tag,
 since=since,
).items():
    try:
        # Add \n escape character to print() to organize tweets
        print('\nTweet by: @' + tweet.user.screen_name)
        print('this is the ',number,' user')
        number += 1
        # Retweet tweets as they are found
        tweet.retweet()
        print('Retweeted the tweet')
        sleep(randint(1,2)) #for the fast retweet
        # sleep(36) 36 seconds for the all day slow retweet
        

    except tweepy.TweepError as e:
        print(e.reason)

    except StopIteration:
        break
    
```

**Comment by Harmon758** [comment, marked as answer]
[`API.search`](https://tweepy.readthedocs.io/en/latest/api.html#API.search) accepts a `since_id` parameter, not `since`.
You should pass the ID of the most recent Tweet you've processed already as `since_id`.
However, this seems like something [using a stream](https://tweepy.readthedocs.io/en/latest/streaming_how_to.html) would be much better suited for, especially if you want it to be realtime.

**Reply by yekmil** [reply]
I don't know how to use a stream in order to retweet tweets containing a keyword. can you elaborate with code?
can you show me an example?

**Reply by Harmon758** [reply]
https://tweepy.readthedocs.io/en/latest/streaming_how_to.html and https://github.com/tweepy/tweepy/blob/master/examples/streaming.py have examples of streaming.
You'll probably want to Retweet within the `on_status` method as you process each Tweet.
